internals/review_models.py

```diff
@@ -0,0 +1,152 @@
+      limit=None):
+    """Return the requested approvals."""
+    query = ApprovalVote.query()
+    if feature_id is not None:
+      query = query.filter(ApprovalVote.feature_id == feature_id)
+    if field_id is not None:
+      query = query.filter(ApprovalVote.field_id == field_id)
+    if states is not None:
+      query = query.filter(ApprovalVote.state.IN(states))
+    if set_by is not None:
+      query = query.filter(ApprovalVote.set_by == set_by)
+    approvals = query.fetch(limit)
+    return approvals
+
+  @classmethod
+  def sorted_by_pending_request_date(cls, descending):
+    """Return feature_ids of pending approvals sorted by request date."""
+    query = ApprovalVote.query(
+        ApprovalVote.state == ApprovalVote.REVIEW_REQUESTED)
+    if descending:
+      query = query.order(-ApprovalVote.set_on)
+    else:
+      query = query.order(ApprovalVote.set_on)
+
+    pending_approvals = query.fetch(projection=['feature_id'])
+    feature_ids = utils.dedupe(pa.feature_id for pa in pending_approvals)
+    return feature_ids
+
+  @classmethod
+  def sorted_by_review_date(cls, descending):
+    """Return feature_ids of reviewed approvals sorted by last review."""
+    query = ApprovalVote.query(ApprovalVote.state.IN(ApprovalVote.FINAL_STATES))
+    if descending:
+      query = query.order(-ApprovalVote.set_on)
+    else:
+      query = query.order(ApprovalVote.set_on)
+    recent_approvals = query.fetch(projection=['feature_id'])
+
+    feature_ids = utils.dedupe(ra.feature_id for ra in recent_approvals)
+    return feature_ids
+
+  @classmethod
+  def is_valid_state(cls, new_state):
+    """Return true if new_state is valid."""
+    return new_state in cls.APPROVAL_VALUES
+
+  @classmethod
+  def set_approval(cls, feature_id, field_id, new_state, set_by_email):
+    """Add or update an approval value."""
+    if not cls.is_valid_state(new_state):
+      raise ValueError('Invalid approval state')
+
+    now = datetime.datetime.now()
+    existing_list = cls.get_approvals(
+        feature_id=feature_id, field_id=field_id, set_by=set_by_email)
+    if existing_list:
+      existing = existing_list[0]
+      existing.set_on = now
+      existing.state = new_state
+      existing.put()
+      return
+
+    new_appr = Approval(
+        feature_id=feature_id, field_id=field_id, state=new_state,
+        set_on=now, set_by=set_by_email)
+    new_appr.put()
+
+  @classmethod
+  def clear_request(cls, feature_id, field_id):
+    """After the review requirement has been satisfied, remove the request."""
+    review_requests = cls.get_approvals(
+        feature_id=feature_id, field_id=field_id, states=[cls.REVIEW_REQUESTED])
+    for rr in review_requests:
+      rr.key.delete()
+
+
+class Amendment(ndb.Model):
+  """Comments can log changes to other fields of the feature."""
+  field_name = ndb.StringProperty()  // from QUERIABLE_FIELDS
+  old_value = ndb.TextProperty()
+  new_value = ndb.TextProperty()
```

Comment:
This seems like it would need a timestamp of when the change was made if that doesn't get generated by default.

Reply:
Every amendment is part of a comment, which has a timestamp.    In cases where the user did not enter any comment content, the comment will be just an empty string.